shared_model/backend/protobuf/query_responses/proto_query_response.hpp

```diff
@@ -85,18 +68,32 @@ namespace shared_model {
           : QueryResponse(std::move(o.proto_)) {}
 
       const QueryResponseVariantType &get() const override {
-        return *variant_;
+        return *ivariant_;
       }
 
       const interface::types::HashType &queryHash() const override {
         return *hash_;
       }
 
      private:
+      template <typename T>
+      using Lazy = detail::LazyInitializer<T>;
+
+      using LazyVariantType = Lazy<ProtoQueryResponseVariantType>;
+
       const LazyVariantType variant_{[this] {
-        return loadQueryResponse<ProtoQueryResponseListType>(*proto_);
+        auto &&ar = *proto_;
+        int which =
+            ar.GetDescriptor()->FindFieldByNumber(ar.response_case())->index();
+        return shared_model::detail::variant_impl<ProtoQueryResponseListType>::
+            template load<ProtoQueryResponseVariantType>(
+                std::forward<decltype(ar)>(ar), which);
       }};
 
+      const Lazy<QueryResponseVariantType> ivariant_{
```

Comment:
Is there any point in that additional variant?

Reply:
The issue is return type of `get()` method. Boost 1.65.1 `boost::apply_visitor` and `boost::variant::apply` don't accept rvalues, so we need a reference. This can be done by storing the variant in class with minimal changes to the project.